File: clients/python/sdk/api/v1/statuses/item/statuses_item_request_builder.py
from __future__ import annotations
from dataclasses import dataclass
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.method import Method
from kiota_abstractions.request_adapter import RequestAdapter
from kiota_abstractions.request_information import RequestInformation
from kiota_abstractions.request_option import RequestOption
from kiota_abstractions.response_handler import ResponseHandler
from kiota_abstractions.serialization import Parsable, ParsableFactory
from typing import Any, Callable, Dict, List, Optional, Union

from .....models import status
from .bookmark import bookmark_request_builder
from .context import context_request_builder
from .favourite import favourite_request_builder
from .favourited_by import favourited_by_request_builder
from .history import history_request_builder
from .mute import mute_request_builder
from .pin import pin_request_builder
from .reblog import reblog_request_builder
from .reblogged_by import reblogged_by_request_builder
from .source import source_request_builder
from .unbookmark import unbookmark_request_builder
from .unfavourite import unfavourite_request_builder
from .unmute import unmute_request_builder
from .unpin import unpin_request_builder
from .unreblog import unreblog_request_builder

class StatusesItemRequestBuilder():
    """
    Builds and executes requests for operations under /api/v1/statuses/{id}
    """
    def bookmark(self) -> bookmark_request_builder.BookmarkRequestBuilder:
        """
        The bookmark property
        """
        return bookmark_request_builder.BookmarkRequestBuilder(self.request_adapter, self.path_parameters)
    
    def context(self) -> context_request_builder.ContextRequestBuilder:
        """
        The context property
        """
        return context_request_builder.ContextRequestBuilder(self.request_adapter, self.path_parameters)
    
    def favourite(self) -> favourite_request_builder.FavouriteRequestBuilder:
        """
        The favourite property
        """
        return favourite_request_builder.FavouriteRequestBuilder(self.request_adapter, self.path_parameters)
    
    def favourited_by(self) -> favourited_by_request_builder.Favourited_byRequestBuilder:
        """
        The favourited_by property
        """
        return favourited_by_request_builder.Favourited_byRequestBuilder(self.request_adapter, self.path_parameters)
    
    def history(self) -> history_request_builder.HistoryRequestBuilder:
        """
        The history property
        """
        return history_request_builder.HistoryRequestBuilder(self.request_adapter, self.path_parameters)
    
    def mute(self) -> mute_request_builder.MuteRequestBuilder:
        """
        The mute property
        """
        return mute_request_builder.MuteRequestBuilder(self.request_adapter, self.path_parameters)
    
    def pin(self) -> pin_request_builder.PinRequestBuilder:
        """
        The pin property
        """
        return pin_request_builder.PinRequestBuilder(self.request_adapter, self.path_parameters)
    
    def reblog(self) -> reblog_request_builder.ReblogRequestBuilder:
        """
        The reblog property
        """
        return reblog_request_builder.ReblogRequestBuilder(self.request_adapter, self.path_parameters)
    
    def reblogged_by(self) -> reblogged_by_request_builder.Reblogged_byRequestBuilder:
        """
        The reblogged_by property
        """
        return reblogged_by_request_builder.Reblogged_byRequestBuilder(self.request_adapter, self.path_parameters)
    
    def source(self) -> source_request_builder.SourceRequestBuilder:
        """
        The source property
        """
        return source_request_builder.SourceRequestBuilder(self.request_adapter, self.path_parameters)
    
    def unbookmark(self) -> unbookmark_request_builder.UnbookmarkRequestBuilder:
        """
        The unbookmark property
        """
        return unbookmark_request_builder.UnbookmarkRequestBuilder(self.request_adapter, self.path_parameters)
    
    def unfavourite(self) -> unfavourite_request_builder.UnfavouriteRequestBuilder:
        """
        The unfavourite property
        """
        return unfavourite_request_builder.UnfavouriteRequestBuilder(self.request_adapter, self.path_parameters)
    
    def unmute(self) -> unmute_request_builder.UnmuteRequestBuilder:
        """
        The unmute property
        """
        return unmute_request_builder.UnmuteRequestBuilder(self.request_adapter, self.path_parameters)
    
    def unpin(self) -> unpin_request_builder.UnpinRequestBuilder:
        """
        The unpin property
        """
        return unpin_request_builder.UnpinRequestBuilder(self.request_adapter, self.path_parameters)
    
    def unreblog(self) -> unreblog_request_builder.UnreblogRequestBuilder:
        """
        The unreblog property
        """
        return unreblog_request_builder.UnreblogRequestBuilder(self.request_adapter, self.path_parameters)
    
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Optional[Union[Dict[str, Any], str]] = None) -> None:
        """
        Instantiates a new StatusesItemRequestBuilder and sets the default values.
        Args:
            pathParameters: The raw url or the Url template parameters for the request.
            requestAdapter: The request adapter to use to execute the requests.
        """
        if path_parameters is None:
            raise Exception("path_parameters cannot be undefined")
        if request_adapter is None:
            raise Exception("request_adapter cannot be undefined")
        # Url template to use to build the URL for the current request builder
        self.url_template: str = "{+baseurl}/api/v1/statuses/{id}"

        url_tpl_params = get_path_parameters(path_parameters)
        self.path_parameters = url_tpl_params
        self.request_adapter = request_adapter
    
    def create_delete_request_information(self,request_configuration: Optional[StatusesItemRequestBuilderDeleteRequestConfiguration] = None) -> RequestInformation:
        request_info = RequestInformation()
        request_info.url_template = self.url_template
        request_info.path_parameters = self.path_parameters
        request_info.http_method = Method.DELETE
        request_info.headers["Accept"] = "application/json"
        if request_configuration:
            request_info.add_request_headers(request_configuration.headers)
            request_info.add_request_options(request_configuration.options)
        return request_info
    
    def create_get_request_information(self,request_configuration: Optional[StatusesItemRequestBuilderGetRequestConfiguration] = None) -> RequestInformation:
        request_info = RequestInformation()
        request_info.url_template = self.url_template
        request_info.path_parameters = self.path_parameters
        request_info.http_method = Method.GET
        request_info.headers["Accept"] = "application/json"
        if request_configuration:
            request_info.add_request_headers(request_configuration.headers)
            request_info.add_request_options(request_configuration.options)
        return request_info
    
    def create_put_request_information(self,body: bytes, request_configuration: Optional[StatusesItemRequestBuilderPutRequestConfiguration] = None) -> RequestInformation:
        if body is None:
            raise Exception("body cannot be undefined")
        request_info = RequestInformation()
        request_info.url_template = self.url_template
        request_info.path_parameters = self.path_parameters
        request_info.http_method = Method.PUT
        request_info.headers["Accept"] = "application/json"
        if request_configuration:
            request_info.add_request_headers(request_configuration.headers)
            request_info.add_request_options(request_configuration.options)
        request_info.set_stream_content(body)
        return request_info
    
    async def delete(self,request_configuration: Optional[StatusesItemRequestBuilderDeleteRequestConfiguration] = None, response_handler: Optional[ResponseHandler] = None) -> Optional[status.Status]:
        request_info = self.create_delete_request_information(
            request_configuration
        )
        if not self.request_adapter:
            raise Exception("Http core is null") 
        return await self.request_adapter.send_async(request_info, status.Status, response_handler, None)
    
    async def get(self,request_configuration: Optional[StatusesItemRequestBuilderGetRequestConfiguration] = None, response_handler: Optional[ResponseHandler] = None) -> Optional[status.Status]:
        request_info = self.create_get_request_information(
            request_configuration
        )
        if not self.request_adapter:
            raise Exception("Http core is null") 
        return await self.request_adapter.send_async(request_info, status.Status, response_handler, None)
    
    async def put(self,body: bytes, request_configuration: Optional[StatusesItemRequestBuilderPutRequestConfiguration] = None, response_handler: Optional[ResponseHandler] = None) -> Optional[status.Status]:
        if body is None:
            raise Exception("body cannot be undefined")
        request_info = self.create_put_request_information(
            body, request_configuration
        )
        if not self.request_adapter:
            raise Exception("Http core is null") 
        return await self.request_adapter.send_async(request_info, status.Status, response_handler, None)
    
    @dataclass
    class StatusesItemRequestBuilderDeleteRequestConfiguration():
        """
        Configuration for the request such as headers, query parameters, and middleware options.
        """
        # Request headers
        headers: Optional[Dict[str, str]] = None

        # Request options
        options: Optional[List[RequestOption]] = None

    
    @dataclass
    class StatusesItemRequestBuilderGetRequestConfiguration():
        """
        Configuration for the request such as headers, query parameters, and middleware options.
        """
        # Request headers
        headers: Optional[Dict[str, str]] = None

        # Request options
        options: Optional[List[RequestOption]] = None

    
    @dataclass
    class StatusesItemRequestBuilderPutRequestConfiguration():
        """
        Configuration for the request such as headers, query parameters, and middleware options.
        """
        # Request headers
        headers: Optional[Dict[str, str]] = None

        # Request options
        options: Optional[List[RequestOption]] = None

    

File: clients/node/sdk/api/v1/statuses/item/statusesItemRequestBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusesItemRequestBuilder = void 0;
const createStatusFromDiscriminatorValue_1 = require("../../../../models/createStatusFromDiscriminatorValue");
const bookmarkRequestBuilder_1 = require("./bookmark/bookmarkRequestBuilder");
const contextRequestBuilder_1 = require("./context/contextRequestBuilder");
const favouriteRequestBuilder_1 = require("./favourite/favouriteRequestBuilder");
const favourited_byRequestBuilder_1 = require("./favourited_by/favourited_byRequestBuilder");
const historyRequestBuilder_1 = require("./history/historyRequestBuilder");
const muteRequestBuilder_1 = require("./mute/muteRequestBuilder");
const pinRequestBuilder_1 = require("./pin/pinRequestBuilder");
const reblogRequestBuilder_1 = require("./reblog/reblogRequestBuilder");
const reblogged_byRequestBuilder_1 = require("./reblogged_by/reblogged_byRequestBuilder");
const sourceRequestBuilder_1 = require("./source/sourceRequestBuilder");
const unbookmarkRequestBuilder_1 = require("./unbookmark/unbookmarkRequestBuilder");
const unfavouriteRequestBuilder_1 = require("./unfavourite/unfavouriteRequestBuilder");
const unmuteRequestBuilder_1 = require("./unmute/unmuteRequestBuilder");
const unpinRequestBuilder_1 = require("./unpin/unpinRequestBuilder");
const unreblogRequestBuilder_1 = require("./unreblog/unreblogRequestBuilder");
const kiota_abstractions_1 = require("@microsoft/kiota-abstractions");
/** Builds and executes requests for operations under /api/v1/statuses/{id} */
class StatusesItemRequestBuilder {
    /** The bookmark property */
    get bookmark() {
        return new bookmarkRequestBuilder_1.BookmarkRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The context property */
    get context() {
        return new contextRequestBuilder_1.ContextRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The favourite property */
    get favourite() {
        return new favouriteRequestBuilder_1.FavouriteRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The favourited_by property */
    get favourited_by() {
        return new favourited_byRequestBuilder_1.Favourited_byRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The history property */
    get history() {
        return new historyRequestBuilder_1.HistoryRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The mute property */
    get mute() {
        return new muteRequestBuilder_1.MuteRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The pin property */
    get pin() {
        return new pinRequestBuilder_1.PinRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The reblog property */
    get reblog() {
        return new reblogRequestBuilder_1.ReblogRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The reblogged_by property */
    get reblogged_by() {
        return new reblogged_byRequestBuilder_1.Reblogged_byRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The source property */
    get source() {
        return new sourceRequestBuilder_1.SourceRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The unbookmark property */
    get unbookmark() {
        return new unbookmarkRequestBuilder_1.UnbookmarkRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The unfavourite property */
    get unfavourite() {
        return new unfavouriteRequestBuilder_1.UnfavouriteRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The unmute property */
    get unmute() {
        return new unmuteRequestBuilder_1.UnmuteRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The unpin property */
    get unpin() {
        return new unpinRequestBuilder_1.UnpinRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The unreblog property */
    get unreblog() {
        return new unreblogRequestBuilder_1.UnreblogRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /**
     * Instantiates a new StatusesItemRequestBuilder and sets the default values.
     * @param pathParameters The raw url or the Url template parameters for the request.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    constructor(pathParameters, requestAdapter) {
        if (!pathParameters)
            throw new Error("pathParameters cannot be undefined");
        if (!requestAdapter)
            throw new Error("requestAdapter cannot be undefined");
        this.urlTemplate = "{+baseurl}/api/v1/statuses/{id}";
        const urlTplParams = (0, kiota_abstractions_1.getPathParameters)(pathParameters);
        this.pathParameters = urlTplParams;
        this.requestAdapter = requestAdapter;
    }
    ;
    createDeleteRequestInformation(requestConfiguration) {
        const requestInfo = new kiota_abstractions_1.RequestInformation();
        requestInfo.urlTemplate = this.urlTemplate;
        requestInfo.pathParameters = this.pathParameters;
        requestInfo.httpMethod = kiota_abstractions_1.HttpMethod.DELETE;
        requestInfo.headers["Accept"] = "application/json";
        if (requestConfiguration) {
            requestInfo.addRequestHeaders(requestConfiguration.headers);
            requestInfo.addRequestOptions(requestConfiguration.options);
        }
        return requestInfo;
    }
    ;
    createGetRequestInformation(requestConfiguration) {
        const requestInfo = new kiota_abstractions_1.RequestInformation();
        requestInfo.urlTemplate = this.urlTemplate;
        requestInfo.pathParameters = this.pathParameters;
        requestInfo.httpMethod = kiota_abstractions_1.HttpMethod.GET;
        requestInfo.headers["Accept"] = "application/json";
        if (requestConfiguration) {
            requestInfo.addRequestHeaders(requestConfiguration.headers);
            requestInfo.addRequestOptions(requestConfiguration.options);
        }
        return requestInfo;
    }
    ;
    createPutRequestInformation(body, requestConfiguration) {
        if (!body)
            throw new Error("body cannot be undefined");
        const requestInfo = new kiota_abstractions_1.RequestInformation();
        requestInfo.urlTemplate = this.urlTemplate;
        requestInfo.pathParameters = this.pathParameters;
        requestInfo.httpMethod = kiota_abstractions_1.HttpMethod.PUT;
        requestInfo.headers["Accept"] = "application/json";
        if (requestConfiguration) {
            requestInfo.addRequestHeaders(requestConfiguration.headers);
            requestInfo.addRequestOptions(requestConfiguration.options);
        }
        requestInfo.setStreamContent(body);
        return requestInfo;
    }
    ;
    delete(requestConfiguration, responseHandler) {
        var _a, _b;
        const requestInfo = this.createDeleteRequestInformation(requestConfiguration);
        return (_b = (_a = this.requestAdapter) === null || _a === void 0 ? void 0 : _a.sendAsync(requestInfo, createStatusFromDiscriminatorValue_1.createStatusFromDiscriminatorValue, responseHandler, undefined)) !== null && _b !== void 0 ? _b : Promise.reject(new Error('request adapter is null'));
    }
    ;
    get(requestConfiguration, responseHandler) {
        var _a, _b;
        const requestInfo = this.createGetRequestInformation(requestConfiguration);
        return (_b = (_a = this.requestAdapter) === null || _a === void 0 ? void 0 : _a.sendAsync(requestInfo, createStatusFromDiscriminatorValue_1.createStatusFromDiscriminatorValue, responseHandler, undefined)) !== null && _b !== void 0 ? _b : Promise.reject(new Error('request adapter is null'));
    }
    ;
    put(body, requestConfiguration, responseHandler) {
        var _a, _b;
        if (!body)
            throw new Error("body cannot be undefined");
        const requestInfo = this.createPutRequestInformation(body, requestConfiguration);
        return (_b = (_a = this.requestAdapter) === null || _a === void 0 ? void 0 : _a.sendAsync(requestInfo, createStatusFromDiscriminatorValue_1.createStatusFromDiscriminatorValue, responseHandler, undefined)) !== null && _b !== void 0 ? _b : Promise.reject(new Error('request adapter is null'));
    }
    ;
}
exports.StatusesItemRequestBuilder = StatusesItemRequestBuilder;
